init.zsh

```diff
@@ -17,6 +17,45 @@ if ! autoload -Uz is-at-least || ! is-at-least "$min_zsh_version"; then
 fi
 unset min_zsh_version
 
+# zprezto convenience updater
+function zprezto-update {
+   function cannot-fast-forward {
+       local STATUS="$1"
+       [[ "${STATUS}" ]] && printf "%s\n" "${STATUS}"
+       printf "Unable to fast-forward the changes. You can fix this by "
+       printf "running\ncd '%s' and then\n'git pull' " "${ZPREZTODIR}"
+       printf "to manually pull and possibly merge in changes\n"
+   }
+   pushd -q
+   cd -- "${ZPREZTODIR}"
+   local orig_branch="$(git symbolic-ref HEAD 2>/dev/null | cut -d '/' -f 3)"
+   if [[ "$orig_branch" == "master" ]]; then
+       git fetch
+       local UPSTREAM=$(git rev-parse '@{u}')
+       local LOCAL=$(git rev-parse @)
+       local REMOTE=$(git rev-parse "$UPSTREAM")
+       local BASE=$(git merge-base @ "$UPSTREAM")
+       if [[ $LOCAL == $REMOTE ]]; then
+           printf "There are no updates.\n"
+       elif [[ $LOCAL == $BASE ]]; then
+           printf "There is an update availible. Trying to pull.\n\n"
+           if git pull --ff-only; then
+               printf "Syncing submodules\n"
+               git submodule update --recursive
+           else
+               cannot-fast-forward
+           fi
+       elif [[ $REMOTE == $BASE ]]; then
+           cannot-fast-forward "Commits in master that aren't in upstream."
+       else
+           cannot-fast-forward "Upstream and local have diverged."
+       fi
+   else
+       printf "zprezto install at '%s' is not on the master branch " "${ZPREZTODIR}"
+       printf "(you're on '%s')\nUnable to automatically update.\n" "${orig_branch}"
+   fi
+   popd -q
```

Comment:
It would be nice to have proper return codes if this fails. `popd` could be moved to an atexit function (`trap 'popd -q' EXIT`) which would make it easier to return without having to copy the popd everywhere.